File: BasicArrayQuestions/MakeBuildingFromArray/MakeBuildingFromArray.java


import java.util.Scanner;

public class MakeBuildingFromArray {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter size of array: ");
        int n = sc.nextInt();

        int[] arr = new int[n];

        System.out.println("Enter elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        makeBuilds(arr, n);
    }

    public static void makeBuilds(int[] arr, int n) {

        int max = Integer.MIN_VALUE;
        for(int val : arr){
            if(val>max){
                max=val;
            }
        }
        for(int floor=max;floor>0;floor--){
            for(int i=0;i<arr.length;i++){
                if(arr[i]>=floor){
                System.out.print("* ");//2characters
                }
                else{
                    System.out.print("  ");//2character space
                }
            }
            System.out.println();

        }
        }
    }

File: BasicArrayQuestions/MakeBuildingFromArray/MakeBuildingFromArray.cpp
#include<iostream>
using namespace std;

void makeBuilds(int arr[],int size){
    int max = 0;
    for(int i=0;i<size;i++){
        if(arr[i]>max){
            max=arr[i];
        }
    }
    //Outer For Loop for moving to next line one by one
    for(int floor=max;floor>0;floor--){
        //This for loop is for printing 
        for(int i=0;i<size;i++){
            if(arr[i]>=floor){
                cout << "* ";
            }
            else{
                cout << "  ";
            }

        }
        cout << endl;
    }
}
int main(){
    int n;
        cout << "Enter size of array: ";
        cin >> n;

        int arr[n];

        cout << "Enter elements:" << endl;
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        makeBuilds(arr,n);
        return 0;
}

    